### PR TITLE
make first letter of error message lowercase

### PR DESCRIPTION
**Pull request by mqliang**
Error strings should not be capitalized https://github.com/golang/go/wiki/CodeReviewComments#error-strings

This PR helps to fix this.

Signed-off-by: mqliang mqliang.zju@gmail.com

### CODE REVIEW

**Comment by stevvooe** [comment]
:clap: 

It is very likely these are hardcoded into a 1000 integration tests.

**Comment by thaJeztah** [comment]
thanks @mqliang! can you squash your commits?

**Comment by mqliang** [comment]
ping @stevvooe @thaJeztah

**Comment by vdemeester** [comment]
@mqliang needs a rebase :sweat_smile:

**Comment by mqliang** [comment]
@vdemeester Rebased. Since this PR involves a lot of places, could you try to expedite the review and merge process to avoid a lot of churns.

**Comment by thaJeztah** [comment]
Also needs to be checked if this doesnt result in "rebase hell" for https://github.com/docker/docker/pull/17924

**Comment by runcom** [comment]
pretty sure I've checked everything with `ag` :smile: LGTM

**Comment by askb** [comment]
Nice, just wondering if this PR/issue is only limited only the `New()` or should `logrus.{Debugf(),Warnf(),Println(),Info(),...}` and similar call "message" needs to be changed to start with a lower-case letter ? Thanks.

**Comment by mqliang** [comment]
@askb Error strings should not be capitalized, since they are usually printed following other contex. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by runcom** [comment]
@mqliang sorry this needs a rebase :(

File: graph/pull_v2.go [hunk 1]
```diff
@@ -502,7 +502,7 @@ func fixManifestLayers(m *schema1.Manifest) error {
 
 	if images[len(images)-1].Parent != "" && !allowBaseParentImage {
 		// Windows base layer can point to a base layer parent that is not in manifest.
-		return errors.New("Invalid parent ID in the base layer of the image.")
+		return errors.New("invalid parent ID in the base layer of the image.")
```
Comment (dmcgowan) [inline graph/pull_v2.go]: missed one `graph/pull_v2.go:505:21: error strings should not end with punctuation`

**Comment by LK4D4** [comment]
@mqliang golint error :/

**Comment by askb** [comment]
@mqliang, I noticed instances where error messages dont get prefixed with a context and is printed. Here for example refer  `vendor/src/github.com/docker/distribution/reference/reference.go`
`+44 ErrReferenceInvalidFormat = errors.New("invalid reference format")`. The error message gets printed as follows on `docker pull`, not sure if this is within the scope of this PR.
```
# docker pull ubuntu:14:04
invalid reference format
```
I think before converting all the error messages to lower case would be applciable in all situations, maybe testing the context of each error helps. Thanks.

**Comment by thaJeztah** [comment]
@mqliang looks like this needs another rebase :cry:

**Comment by stevvooe** [comment]
@askb In Go, the creation of symbolic errors is helpful for quickly detecting and responding to error conditions. This let's us build code that actually handles error conditions:


```go
if err == ErrInvalidReference {
    // go get a valid reference
}
```

Typically, such errors should be decorated in the caller, where the identity of the error no longer matters. For example, there should be places that do this:

```
if err := InvalidReference(ref); err != nil {
    // calling scope no longer cares about invalid references
    return fmt.Errorf("%v: %v", ref, err)
}
```

**Comment by stevvooe** [comment]
@askb With that said, I'm not sure that the context of errors is at all relevant to this particular PR.

**Comment by askb** [comment]
@stevvooe thanks, sorry for any confusion :)

**Pull request closed without merging**